src/socket.cpp

```diff
@@ -95,14 +95,14 @@ int gai_getsockaddr(struct sockaddr_storage* ss, const char* host,
 void sockaddr_update_port(struct sockaddr_storage* ss, short port)
 {
     switch (ss->ss_family) {
-    case AF_INET:
-        _RCAST(struct sockaddr_in*, ss)->sin_port = htons(port);
-        break;
-    case AF_INET6:
-        _RCAST(struct sockaddr_in6*, ss)->sin6_port = htons(port);
-        break;
-    default:
-        ERROR("Unsupported family type");
+        case AF_INET:
```

Comment:
I'm not fond of indented case myself.
And I prefer python-rules for indenting: 4 spaces unless it conflicts with readability: after `if (` you take 8 spaces so the if-condition stands out from the if-body.

Did you use some auto-indent on this file? It didn't work out so well at the end ;)